src/controllers/controller.doughnut.js

```diff
@@ -46,20 +46,14 @@ defaults._set('doughnut', {
 				if (data.labels.length && data.datasets.length) {
 					return data.labels.map(function(label, i) {
 						var meta = chart.getDatasetMeta(0);
-						var ds = data.datasets[0];
-						var arc = meta.data[i];
-						var custom = arc && arc.custom || {};
-						var arcOpts = chart.options.elements.arc;
-						var fill = resolve([custom.backgroundColor, ds.backgroundColor, arcOpts.backgroundColor], undefined, i);
-						var stroke = resolve([custom.borderColor, ds.borderColor, arcOpts.borderColor], undefined, i);
-						var bw = resolve([custom.borderWidth, ds.borderWidth, arcOpts.borderWidth], undefined, i);
+						var style = meta.controller.getStyle(i);
```

Comment:
I wonder what the performance impact of this will be. Before we were only resolving three properties and now we're resolving seven

I also noticed that in `_resolveElementOptions` we're resolving the hover properties, but then I don't think we're ever using them. `setHoverStyle` seems to do the resolution again and ignores the previously resolved styles. Perhaps we should not be trying to resolve them in `_resolveElementOptions`

Reply:
Maybe we can use cached `element._options` if the element exists since element updates are done prior to layout updates. Some controllers are currently not using `_options`, so we need to update them.

Regarding `setHoverStyle`, most of controllers other than the bar controller are using cached styles, and resolving on update is more efficient than doing on hover event. Further optimization may be possible, though (eg. `valueOrDefault(options.hoverBackgroundColor, getHoverColor(options.backgroundColor))` can be done in `_resolveElementOptions`).

Reply:
Updated the code to use `element._options` if available.